crates/edr_coverage/src/collector.rs

```diff
@@ -0,0 +1,36 @@
+use edr_eth::{Bytes, HashSet};
+use edr_evm::{inspector::Inspector, interpreter::InterpreterTypes};
+
+use crate::COVERAGE_ADDRESS;
+
+#[derive(Default)]
+pub struct CoverageHitCollector {
+    hits: HashSet<Bytes>,
+}
+
+impl CoverageHitCollector {
+    /// Returns the collected coverage hits.
+    pub fn into_hits(self) -> HashSet<Bytes> {
+        self.hits
+    }
+
+    fn record_hit(&mut self, hit: Bytes) {
+        self.hits.insert(hit);
+    }
+}
+
+impl<ContextT, InterpreterT: InterpreterTypes> Inspector<ContextT, InterpreterT>
+    for CoverageHitCollector
+{
+    fn call(
+        &mut self,
+        _context: &mut ContextT,
+        inputs: &mut edr_evm::interpreter::CallInputs,
+    ) -> Option<edr_evm::interpreter::CallOutcome> {
+        if inputs.bytecode_address == COVERAGE_ADDRESS {
+            self.record_hit(inputs.input.clone());
```

Comment:
Is it correct to record the entire call input data here, as opposed to extracting the `coverageId` from it and recording just that? I would've expected the latter.

Reply:
Ah, I see that I was wrong. The call that is being recorded is not `Coverage.sendHit`, but the one constructed here:

https://github.com/NomicFoundation/edr/blob/1ce8240a83511baa1fe820a77cee0490669fea7f/data/contracts/coverage.sol#L15-L16